### PR TITLE
Plugin geomap/aoi: Allow to position the AOI buttons below the coordinates fields and add tooltips to AOI buttons

### PR DESCRIPTION
**Pull request by josegar74**
A new setting has been added to allow to display the AOI Add/Clear buttons below the coordinate fields, so these fields can have more space to display/edit the coordinate values. 

Also tooltips have been added to the buttons.

To display the buttons below the coordinates, add the setting `"buttonsAtBottom": true` to the geomap configuration:

```
data-wb-geomap='{
             "aoi": { "toggle": false, "buttonsAtBottom": true },
```

aoi-buttons-bottom

If not specified the setting or set as false, works as default, displaying the coordinates and buttons in the same row:

```
data-wb-geomap='{
             "aoi": { "toggle": false },
```

aoi-buttons-default

### CODE REVIEW

**Comment by LaurentGoderre** [comment]
The changes to the .csv will be overwritten since they come from an online spreadsheet. @jvanulde are you able to add those strings to the spreadsheet?

**Comment by LaurentGoderre** [comment]
The build failed because we use the jQuery code style guide.

File: src/plugins/deps/geomap-lib.js [hunk 1]
```diff
@@ -2145,40 +2151,42 @@ var componentName = "wb-geomap",
...
 						"<label for='geomap-aoi-minx-" + geomap.mapid + "' class='wb-inv'>" + i18nText.aoiWest + "</label>" +
 						"<div class='input-group input-group-sm'>" +
 							"<span class='input-group-addon'>" + i18nText.aoiWest.charAt( 0 ) + "</span>" +
 							"<input type='number' id='geomap-aoi-minx-" + geomap.mapid + "' placeholder='-180' class='form-control input-sm' min='-180' max='180' step='0.000001'></input> " +
 						"</div>" +
-					"</div>" +
-					"<div class='col-md-4'>" +
-						"<button class='btn btn-default btn-sm' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +
-						"<button class='btn btn-default btn-sm' id='geomap-aoi-btn-clear-" + geomap.mapid + "'>" + i18nText.aoiBtnClear + "</button> " +
+					"</div>" + (geomap.aoiButtonsAtBottom?"</div><div class='row'>":"") +
```
Comment (LaurentGoderre) [inline src/plugins/deps/geomap-lib.js]: add space in the parenthesis `( geomap.aoiButtonsAtBottom ? "</div><div class='row'>" : "" )`

File: src/plugins/deps/geomap-lib.js [hunk 2]
```diff
@@ -2145,40 +2151,42 @@ var componentName = "wb-geomap",
...
+					"<div class='" + (geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4") + "'>" +
```
Comment (LaurentGoderre) [inline src/plugins/deps/geomap-lib.js]: `( geomap.aoiButtonsAtBottom ? "col-md-12" : "col-md-4" )`

**Comment by jvanulde** [comment]
I don't think anything needs to be changed in the spreadsheet. As well we need to add some space between the coordinate inputs and the buttons.

**Comment by LaurentGoderre** [comment]
These strings already exist in the i18n.csv?

**Comment by jvanulde** [comment]
Which strings? As far as I can tell new strings have not been added.

File: src/i18n/i18n.csv [hunk 3]
```diff
@@ -193,6 +193,9 @@ Geomap,geo-mpctrl,Map control,Contrôle de la carte,Map beheer,Harta e kontrolli
...
 ,geo-aoi-btndraw,Draw,Tracer,Teken,Barazim,رسم,Նկարել,Равенство,畫,画,Kreslit,Trekken,Joonistama,Ziehen,Κλήρωση,खींचना,Rajzol,Draga,Menarik,Disegnare,描く,무승부,Izdarīt,Atkreipti,Rysować,Desenhar,Desenhar,Рисовать,Kresliť,Dibujar,วาด,Çizmek,Малювати,Vẽ
 ,geo-aoi-btnclear,Clear,Enlever,Verwyder,Heq,نزع,Հեռացնել,Премахнете,清除,清除,Odstranit,Verwijderen,Eemaldama,Entfernen,Κατάργηση,हटाना,Eltávolít,Fjarlægja,Menghapus,Rimuovere,削除する,제거,Noņemt,Pašalinti,Usunąć,Remover,Remover,Удалять,Odstrániť,Eliminar,เอาออก,Kaldırmak,Видаляти,Hủy bỏ
+,geo-aoi-btndraw-tip,Enable/Disable Draw Mode,Activer/désactiver le mode dessin,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
+,geo-aoi-btnclear-tip,Clear Coordinates and Bounding Box,Effacer les coordonnées et la boîte englobante,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
+,geo-aoi-btnadd-tip,Update the bounding box according to new coordinates,Mettre à jour la boîte englobante selon les nouvelles coordonnées,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```
Comment (LaurentGoderre) [inline src/i18n/i18n.csv]: @jvanulde those strings

**Comment by josegar74** [comment]
I'll update the code with suggestions. For me was building locally, not clear whats the difference

**Comment by jvanulde** [comment]
Merci @josegar74

**Comment by LaurentGoderre** [comment]
Some checks are only run on the full build to reduce the build time

**Comment by jvanulde** [comment]
@LaurentGoderre strings have been added to spreadsheet.

**Comment by LaurentGoderre** [comment]
@josegar74 can you run `grunt update-i18n` and ammend the result to your commit?

**Comment by pjackson28** [comment]
Please squash your commits (using the format Geomap: Allow to position the AOI..." and fix the following errors flagged by the build:
```
Running "jscs:all" (jscs) task

Missing space after opening round bracket at src/plugins/deps/geomap-lib.js :

  2152 |   geomProj;

  2153 |

  2154 |  var classCoordName = (geomap.aoiButtonsAtBottom)?"'col-md-3'":"'col-md-2'";

--------------------------------^

  2155 |

  2156 |  aoiDiv.append( "<fieldset id='form-aoi-" + geomap.mapid + "'>" +

Missing space before closing round bracket at src/plugins/deps/geomap-lib.js :

  2152 |   geomProj;

  2153 |

  2154 |  var classCoordName = (geomap.aoiButtonsAtBottom)?"'col-md-3'":"'col-md-2'";

---------------------------------------------------------^

  2155 |

  2156 |  aoiDiv.append( "<fieldset id='form-aoi-" + geomap.mapid + "'>" +

Missing space after test at src/plugins/deps/geomap-lib.js :

  2152 |   geomProj;

  2153 |

  2154 |  var classCoordName = (geomap.aoiButtonsAtBottom)?"'col-md-3'":"'col-md-2'";

----------------------------------------------------------^

  2155 |

  2156 |  aoiDiv.append( "<fieldset id='form-aoi-" + geomap.mapid + "'>" +

Missing space before consequent at src/plugins/deps/geomap-lib.js :

  2152 |   geomProj;

  2153 |

  2154 |  var classCoordName = (geomap.aoiButtonsAtBottom)?"'col-md-3'":"'col-md-2'";

-----------------------------------------------------------^

  2155 |

  2156 |  aoiDiv.append( "<fieldset id='form-aoi-" + geomap.mapid + "'>" +

Missing space after consequent at src/plugins/deps/geomap-lib.js :

  2152 |   geomProj;

  2153 |

  2154 |  var classCoordName = (geomap.aoiButtonsAtBottom)?"'col-md-3'":"'col-md-2'";

-----------------------------------------------------------------------^

  2155 |

  2156 |  aoiDiv.append( "<fieldset id='form-aoi-" + geomap.mapid + "'>" +

Missing space before alternate at src/plugins/deps/geomap-lib.js :

  2152 |   geomProj;

  2153 |

  2154 |  var classCoordName = (geomap.aoiButtonsAtBottom)?"'col-md-3'":"'col-md-2'";

------------------------------------------------------------------------^

  2155 |

  2156 |  aoiDiv.append( "<fieldset id='form-aoi-" + geomap.mapid + "'>" +

Missing space after test at src/plugins/deps/geomap-lib.js :

  2184 |       "<input type='number' id='geomap-aoi-minx-" + geomap.mapid + "' placeholder='-180' class='form-control input-sm' min='-180' max='180' step='0.000001'></input> " +

  2185 |      "</div>" +

  2186 |     "</div>" + ( geomap.aoiButtonsAtBottom?"</div><div class='row'>":"" ) +

---------------------------------------------------^

  2187 |     "<div class='" + ( geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4" ) + "'>" +

  2188 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnAddTip + "' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +

Missing space before consequent at src/plugins/deps/geomap-lib.js :

  2184 |       "<input type='number' id='geomap-aoi-minx-" + geomap.mapid + "' placeholder='-180' class='form-control input-sm' min='-180' max='180' step='0.000001'></input> " +

  2185 |      "</div>" +

  2186 |     "</div>" + ( geomap.aoiButtonsAtBottom?"</div><div class='row'>":"" ) +

----------------------------------------------------^

  2187 |     "<div class='" + ( geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4" ) + "'>" +

  2188 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnAddTip + "' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +

Missing space after consequent at src/plugins/deps/geomap-lib.js :

  2184 |       "<input type='number' id='geomap-aoi-minx-" + geomap.mapid + "' placeholder='-180' class='form-control input-sm' min='-180' max='180' step='0.000001'></input> " +

  2185 |      "</div>" +

  2186 |     "</div>" + ( geomap.aoiButtonsAtBottom?"</div><div class='row'>":"" ) +

-----------------------------------------------------------------------------^

  2187 |     "<div class='" + ( geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4" ) + "'>" +

  2188 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnAddTip + "' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +

Missing space before alternate at src/plugins/deps/geomap-lib.js :

  2184 |       "<input type='number' id='geomap-aoi-minx-" + geomap.mapid + "' placeholder='-180' class='form-control input-sm' min='-180' max='180' step='0.000001'></input> " +

  2185 |      "</div>" +

  2186 |     "</div>" + ( geomap.aoiButtonsAtBottom?"</div><div class='row'>":"" ) +

------------------------------------------------------------------------------^

  2187 |     "<div class='" + ( geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4" ) + "'>" +

  2188 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnAddTip + "' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +

Missing space after test at src/plugins/deps/geomap-lib.js :

  2185 |      "</div>" +

  2186 |     "</div>" + ( geomap.aoiButtonsAtBottom?"</div><div class='row'>":"" ) +

  2187 |     "<div class='" + ( geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4" ) + "'>" +

---------------------------------------------------------^

  2188 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnAddTip + "' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +

  2189 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnClearTip + "' id='geomap-aoi-btn-clear-" + geomap.mapid + "'>" + i18nText.aoiBtnClear + "</button> " +

Missing space before consequent at src/plugins/deps/geomap-lib.js :

  2185 |      "</div>" +

  2186 |     "</div>" + ( geomap.aoiButtonsAtBottom?"</div><div class='row'>":"" ) +

  2187 |     "<div class='" + ( geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4" ) + "'>" +

----------------------------------------------------------^

  2188 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnAddTip + "' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +

  2189 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnClearTip + "' id='geomap-aoi-btn-clear-" + geomap.mapid + "'>" + i18nText.aoiBtnClear + "</button> " +

Missing space after consequent at src/plugins/deps/geomap-lib.js :

  2185 |      "</div>" +

  2186 |     "</div>" + ( geomap.aoiButtonsAtBottom?"</div><div class='row'>":"" ) +

  2187 |     "<div class='" + ( geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4" ) + "'>" +

---------------------------------------------------------------------^

  2188 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnAddTip + "' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +

  2189 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnClearTip + "' id='geomap-aoi-btn-clear-" + geomap.mapid + "'>" + i18nText.aoiBtnClear + "</button> " +

Missing space before alternate at src/plugins/deps/geomap-lib.js :

  2185 |      "</div>" +

  2186 |     "</div>" + ( geomap.aoiButtonsAtBottom?"</div><div class='row'>":"" ) +

  2187 |     "<div class='" + ( geomap.aoiButtonsAtBottom?"col-md-12":"col-md-4" ) + "'>" +

----------------------------------------------------------------------^

  2188 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnAddTip + "' id='geomap-aoi-btn-draw-" + geomap.mapid + "'>" + i18nText.add + "</button> " +

  2189 |      "<button class='btn btn-default btn-sm' title='" + i18nText.aoiBtnClearTip + "' id='geomap-aoi-btn-clear-" + geomap.mapid + "'>" + i18nText.aoiBtnClear + "</button> " +

>> 14 code style errors found!
```

**Comment by andre-gh** [comment]
Can this pull request be closed, has it gone stale?

**Comment by andre-gh** [comment]
Due to going stale this pull request has been closed.  Please feel free to re-open and comment should you wish to continue development.

**Pull request closed without merging**